client/driver/lxc.go

```diff
@@ -210,9 +210,20 @@ func (d *LxcDriver) Prestart(*ExecContext, *structs.Task) (*PrestartResponse, er
 
 // Start starts the LXC Driver
 func (d *LxcDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse, error) {
+	sresp, err, errCleanup := d.StartWithCleanup(ctx, task)
+	if err != nil {
+		if cleanupErr := errCleanup(); cleanupErr != nil {
+			d.logger.Printf("[ERR] error occured:\n%v\nwhile cleaning up from error in Start: %v", cleanupErr, err)
+		}
+	}
+	return sresp, err
+}
+
+func (d *LxcDriver) StartWithCleanup(ctx *ExecContext, task *structs.Task) (*StartResponse, error, func() error) {
```

Comment:
Minor nitpick, but let's lowercase `Start` so this method isn't exported as it shouldn't be called directly.